gio/runner.go

```diff
@@ -47,16 +47,22 @@ func (runner *gleamRunner) runMapperReducer() {
 		}()
 	}
 
-	stat.FlowHashCode = uint32(runner.Option.HashCode)
-	stat.Stats = []*pb.InstructionStat{
+	stat.Lock()
+	stat.v.FlowHashCode = uint32(runner.Option.HashCode)
+	stat.v.Stats = []*pb.InstructionStat{
 		{
 			StepId: int32(runner.Option.StepId),
 			TaskId: int32(runner.Option.TaskId),
 		},
 	}
+	stat.Unlock()
 
 	if runner.Option.Mapper != "" {
-		if fn, ok := mappers[runner.Option.Mapper]; ok {
```

Comment:
The entries in mappers and reducers are added during init(). Unlikely there would be race conditions.

Reply:
removed